pkg/jobmanager/status.go

```diff
@@ -38,131 +46,183 @@ var TargetRoutingEvents = []event.Name{
 	target.EventTargetInErr,
 }
 
+// RunCoordinates collects information to identify the run for which we want to rebuild the status
+type RunCoordinates struct {
+	JobID types.JobID
+	RunID types.RunID
+}
+
+// TestCoordinates collects information to identify the test for which we want to rebuild the status
+type TestCoordinates struct {
+	RunCoordinates
+	TestName string
+}
+
+// TestStepCoordinates collects information to identify the test step for which we want to rebuild the status
+type TestStepCoordinates struct {
+	TestCoordinates
+	TestStepName  string
+	TestStepLabel string
+}
+
 // buildTargetStatus populates a TestStepStatus object with TestStepStatus information
-func (jm *JobManager) buildTargetStatus(jobID types.JobID, currentTestStepStatus *job.TestStepStatus) error {
+func (jm *JobManager) buildTargetStatuses(coordinates TestStepCoordinates) ([]job.TargetStatus, error) {
 
 	// Fetch all the events associated to Targets routing
 	routingEvents, err := jm.testEvManager.Fetch(
 		[]testevent.QueryField{
-			testevent.QueryJobID(jobID),
+			testevent.QueryJobID(coordinates.JobID),
+			testevent.QueryTestName(coordinates.TestName),
+			testevent.QueryTestStepLabel(coordinates.TestStepLabel),
 			testevent.QueryEventNames(TargetRoutingEvents),
 		},
 	)
 	if err != nil {
-		return fmt.Errorf("could not fetch events associated to target routing: %v", err)
+		return nil, fmt.Errorf("could not fetch events associated to target routing: %v", err)
 	}
 
-	// Filter only the events emitted by this TestStep, group events by Target
+	targetStatuses := make([]job.TargetStatus, 0)
```

Comment:
Just out of curiosity: why this way instead of just? --
```
var targetStatuses []job.TargetStatus
```

Is there any reason to get a non-nil value in case of an empty slice?